### PR TITLE
Ped Spawner, Drunkness, NPC Ignore, Ragdoll, and Fix Metric Blocking

### PR DESCRIPTION
**Pull request by Rxann**
Things I added:
Ped Spawner - spawn all of the peds in the game that I could find.
Drunkness - Applies drunken effect
NPC Ignore - Use natives to have npcs completely ignore you.
Ragdoll - Ragdoll anyone in the lobby.
Fix Metric Blocking - return original function instead of true because returning true was not actually sending the metrics(to my knowledge)

### CODE REVIEW

**Comment by Rxann** [comment]
conflicts have been resolved

File: src/game/features/players/toxic/Ragdoll.cpp
Comment (DayibBaba) [inline src/game/features/players/toxic/Ragdoll.cpp:16]: Does this work on player peds?

File: src/game/features/self/Drunk.cpp
Comment (DayibBaba) [inline src/game/features/self/Drunk.cpp:27]: Use `_GET_PED_DRUNKNESS` to determine whether `_SET_PED_DRUNKNESS` needs to be executed. A simple if check should suffice.

File: src/util/PedModels.hpp
Comment (DayibBaba) [inline src/util/PedModels.hpp:13]: Not a big fan of embedding such a large structure when we could use maybe a json file. But seen as we don't really have data fetching like on Gta, I might let this slide for now.

File: src/util/SpawnPed.cpp
Comment (DayibBaba) [inline src/util/SpawnPed.cpp:12]: use `IS_MODEL_VALID` to verify the model.

File: src/util/SpawnPed.cpp
Comment (DayibBaba) [inline src/util/SpawnPed.cpp:16]: Make this either a for loop or while loop to allow a longer window for the game to stream in the models. Integrate a timeout in case it breaks somehow. Coupled with `IS_MODEL_VALID` this should just work fine, but you never know.

File: src/util/SpawnPed.cpp
Comment (DayibBaba) [inline src/util/SpawnPed.cpp:23]: Why are you randomizing the heading?

File: src/util/SpawnPed.cpp
Comment (DayibBaba) [inline src/util/SpawnPed.cpp:45]: I prefer guard clauses to make the code more readable.

**Review by DayibBaba — CHANGES_REQUESTED** [review]
Not sure about the SendMetric change, need @maybegreat48 to check that one.

File: src/game/features/players/toxic/Ragdoll.cpp
Comment (DayibBaba) [reply src/game/features/players/toxic/Ragdoll.cpp:16]: I bet there is a net event that does work, just like on gta.

File: src/util/PedModels.hpp
Comment (Rxann) [reply src/util/PedModels.hpp:13]: Yes, as I know of there are no methods to fetch data like there is in gta.

File: src/util/SpawnPed.cpp
Comment (Rxann) [reply src/util/SpawnPed.cpp:23]: Just a dynamic element, idk I thought it would be interesting. Should I change it to the players heading?

Comment (DayibBaba) [reply src/util/SpawnPed.cpp:23]: Anything but random

File: src/util/SpawnPed.cpp
Comment (Rxann) [reply src/util/SpawnPed.cpp:12]: I already check that if it is in the cd image. the validity should be checked first is what you are saying right?

File: src/game/features/players/toxic/Ragdoll.cpp
Comment (Rxann) [reply src/game/features/players/toxic/Ragdoll.cpp:16]: There is a net event, but do we have a way to send the event currently?

File: src/util/SpawnPed.cpp
Comment (DayibBaba) [reply src/util/SpawnPed.cpp:12]: Overlooked it.

File: src/game/features/players/toxic/Ragdoll.cpp
Comment (DayibBaba) [reply src/game/features/players/toxic/Ragdoll.cpp:16]: > There is a net event, but do we have a way to send the event currently?

No, but since this won't do anything having it now has no use.

Comment (Rxann) [reply src/game/features/players/toxic/Ragdoll.cpp:16]: alright i will remove the rag doll feature

File: src/util/SpawnPed.cpp
Comment (Rxann) [reply src/util/SpawnPed.cpp:15]: a while loop and for loop do not work in a better manner here. I have tested it and the if statement works better.

**Comment by Rxann** [comment]
Changes have been implemented. I said this on the review comment, but an if statement works better when requesting the model compared to a for or while loop. I have tested it. A block all telemetry option has also been implemented.

**Review by DayibBaba — APPROVED** [review]
Will merge once @maybegreat48 approves.

**Pull request merged**
